File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/breakr/Break.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.data.Int32 import Int32

@PythonTaskInfo
class Break(PythonTask):

    def __init_ports__(self):
        self.in_1 = InputPort(name="In", data_type=Int32)

    def execute(self, grp):
        i = self.in_1.get_data(Int32).values[0]
        if i == 3:
            self.exit_runner()

File: com.systemincloud.examples.tasks.rtask/src/test/r/breakr/Break.R

library(R6)
library(sicrtask)

Break <- R6Class("Break",
  inherit = RTask,
  public = list(
    rtaskinfo = RTaskInfo$new(),
    in1 = NA,
    initialize_ports = function() {
      self$in1 <- InputPort$new(name = "in1", data_type = "Int32")
    },
    initialize = function() {
      self$initialize_ports()
    },
    execute = function(grp) {
	  data <- self$in1$get_data("Int32")
	  i <- data$values[0]
	  if( i == 3 ) self$exit_runner()
	}
  ),
  private = list(
  )
)
